pennylane/qnode.py

```diff
@@ -975,7 +975,10 @@ def __call__(self, *args, **kwargs) -> qml.typing.Result:
 
         # convert result to the interface in case the qfunc has no parameters
 
-        if len(self.tape.get_parameters(trainable_only=False)) == 0:
+        if (
+            len(self.tape.get_parameters(trainable_only=False)) == 0
+            and not self.transform_program.is_informative
+        ):

```

Comment:
Should interface specific stuff be handled by the transforms themselves then? For example, with this change, `qml.matrix` would always return a numpy matrix if applied to a QNode with no parameters, so we would have to add logic to the transform to get the interface from the QNode with the custom qnode transform.

Reply:
It would be nice to keep the responsibility in the qnode, maybe we can add a check on what is the results and if it is not any tensor/list, we skip it.